src/backend/nodes/analysis.py

```
@@ -0,0 +1,334 @@
+            "exec_result": shared.get("exec_result")
+            or {
+                "csv": shared.get("csv_exec_result"),
+                "api": shared.get("api_exec_result"),
+            },
+            "question": shared["question"],
+            "entity_map": shared.get("entity_map", {}),
+            "entities": shared.get("entities", []),
+            "cross_validation": shared.get("cross_validation", {}),
+        }
+
+    def _check_data_completeness(self, exec_result, entities):
```

Ruff (0.14.10)

Missing return type annotation for private function `_check_data_completeness`

(ANN202)